### ISSUE TITLE
subsection new files are not shown in web tailon after server instantiated 

### ISSUE THREAD

**Issue by vkosuri** [gvalkov/tailon-legacy#23]
@gvalkov Thanks for the tool. 

I have using this tool from past 2 week it is quite amazing, but we are using sub-sections to tail the files, This sub-section always creating new files on demand, however the tailon only showing the files which were there before starting web server.  

After the server instantiated if any new file was created it won't shown up in web tailon.

If possible could you please add this feature to exsisisting one.

For example if any file created after logrotate the file won't shown in web tailon

``` yaml

- 'cron':             # it's possible to add sub-sections$
     - '/var/log/cron*'$
```

**Comment by gvalkov** [comment]
Hello @vkosuri. I'm happy to hear that tailon is useful to you! 

This was originally requested in #18 and implemented in 73ddc6daae573f03412c5530895279c28843824b. I just released tailon 0.5.0, which should include this feature. 

Please let me know if you run into any trouble with the new release.

Thanks,
G.

**Comment by vkosuri** [comment]
Thanks @gvalkov after upgradte also it's not working. However when i start the server it working fine. But not showing subsections and auto update directory both.

``` Shell
venv)vkosuri@testing:~/weblog$ tailon -c config.yml
[+68.890095][ INFO] Listening on 0.0.0.0:8181
```

And also i observed other issues like taiion --help and tailon

(venv)vkosuri:~/weblog$ tailon --help

``` Shell
Traceback (most recent call last):
  File "/home/vkosuri/weblog/venv/bin/tailon", line 11, in <module>
    sys.exit(main())
  File "/home/vkosuri/weblog/venv/local/lib/python2.7/site-packages/tailon/main.py", line 183, in main
    parser, opts = parseopts()
  File "/home/vkosuri/weblog/venv/local/lib/python2.7/site-packages/tailon/main.py", line 153, in parseopts
    return parser, parser.parse_args(args)
  File "/home/vkosuri/weblog/venv/local/lib/python2.7/site-packages/tailon/argparse.py", line 1725, in parse_args
    args, argv = self.parse_known_args(args, namespace)
  File "/home/vkosuri/weblog/venv/local/lib/python2.7/site-packages/tailon/argparse.py", line 1754, in parse_known_args
    namespace, args = self._parse_known_args(args, namespace)
  File "/home/vkosuri/weblog/venv/local/lib/python2.7/site-packages/tailon/argparse.py", line 1960, in _parse_known_args
    start_index = consume_optional(start_index)
  File "/home/vkosuri/weblog/venv/local/lib/python2.7/site-packages/tailon/argparse.py", line 1900, in consume_optional
    take_action(action, args, option_string)
  File "/home/vkosuri/weblog/venv/local/lib/python2.7/site-packages/tailon/argparse.py", line 1828, in take_action
    action(self, namespace, argument_values, option_string)
  File "/home/vkosuri/weblog/venv/local/lib/python2.7/site-packages/tailon/argparse.py", line 1021, in __call__
    parser.print_help()
  File "/home/vkosuri/weblog/venv/local/lib/python2.7/site-packages/tailon/argparse.py", line 2358, in print_help
    self._print_message(self.format_help(), file)
  File "/home/vkosuri/weblog/venv/local/lib/python2.7/site-packages/tailon/argparse.py", line 2325, in format_help
    formatter.add_arguments(action_group._group_actions)
  File "/home/vkosuri/weblog/venv/local/lib/python2.7/site-packages/tailon/argparse.py", line 300, in add_arguments
    self.add_argument(action)
  File "/home/vkosuri/weblog/venv/local/lib/python2.7/site-packages/tailon/argparse.py", line 285, in add_argument
    invocations = [get_invocation(action)]
  File "/home/vkosuri/weblog/venv/local/lib/python2.7/site-packages/tailon/utils.py", line 26, in _format_action_invocation
    default = self._get_default_metavar_for_optional(action)
AttributeError: 'CompactHelpFormatter' object has no attribute '_get_default_metavar_for_optional'
```

(venv)vkosuri:~/weblog$ tailon

``` Shell
Traceback (most recent call last):
  File "/home/vkosuri/weblog/venv/bin/tailon", line 11, in <module>
    sys.exit(main())
  File "/home/vkosuri/weblog/venv/local/lib/python2.7/site-packages/tailon/main.py", line 186, in main
    parser.print_help()
  File "/home/vkosuri/weblog/venv/local/lib/python2.7/site-packages/tailon/argparse.py", line 2358, in print_help
    self._print_message(self.format_help(), file)
  File "/home/vkosuri/weblog/venv/local/lib/python2.7/site-packages/tailon/argparse.py", line 2325, in format_help
    formatter.add_arguments(action_group._group_actions)
  File "/home/vkosuri/weblog/venv/local/lib/python2.7/site-packages/tailon/argparse.py", line 300, in add_arguments
    self.add_argument(action)
  File "/home/vkosuri/weblog/venv/local/lib/python2.7/site-packages/tailon/argparse.py", line 285, in add_argument
    invocations = [get_invocation(action)]
  File "/home/vkosuri/weblog/venv/local/lib/python2.7/site-packages/tailon/utils.py", line 26, in _format_action_invocation
    default = self._get_default_metavar_for_optional(action)
AttributeError: 'CompactHelpFormatter' object has no attribute '_get_default_metavar_for_optional'
```

**Comment by gvalkov** [comment]
Thanks for the catch! I just pushed 0.5.1 with a fix for the `--help` option.

The auto-updating functionality works only with directories. Do you mind sharing your config.yaml?

**Comment by vkosuri** [comment]
@gvalkov Thanks it's working on directories,

Once again thank you very much. I am closing this issue hence the problem solved with `0.5.1`
